integration/test_timescaledb.py

```
@@ -98,69 +95,67 @@ def timescaledb(request: "pytest.FixtureRequest") -> Iterator[TimescaleDBData]:
+                    varchar_col VARCHAR(100)
+                );
+                """
+            )
+
+            cursor.execute(f"SELECT create_hypertable('{table_name}', 'time');")
+
+        columns_config = [
+            "binary_col",
+            "bool_col",
+            "double_col",
+            "float_col",
+            "int_col",
+            "timestamp_col",
+            "nchar_col",
+            "varchar_col",
+        ]
 
-    test_cursor.execute(f"SELECT create_hypertable('{table_name}', 'time');")
-    test_cursor.close()
-
-    columns_config = [
-        "binary_col",
-        "bool_col",
-        "double_col",
-        "float_col",
-        "int_col",
-        "timestamp_col",
-        "nchar_col",
-        "varchar_col",
-    ]
-
-    yield (
-        test_connection,
-        conn_host,
-        conn_port,
-        conn_database,
-        conn_user,
-        conn_password,
-        table_name,
-        dsn,
-        timestamp_precision,
-        columns_config,
-    )
+        yield (
+            test_connection,
+            conn_host,
+            conn_port,
+            conn_database,
+            conn_user,
+            conn_password,
+            table_name,
+            dsn,
+            timestamp_precision,
+            columns_config,
+        )
 
-    # Cleanup
-    try:
-        cleanup_cursor = test_connection.cursor()
-        cleanup_cursor.execute(f"DROP TABLE IF EXISTS {table_name} CASCADE;")
-        cleanup_cursor.close()
-    except Exception as e:
-        print(f"Cleanup error: {e}")
-    test_connection.close()
+    finally:
+        # Cleanup - guaranteed to run even if test fails
+        try:
+            with test_connection.cursor() as cleanup_cursor:
+                cleanup_cursor.execute(f"DROP TABLE IF EXISTS {table_name} CASCADE;")
+            # Close connection after cursor operations are complete
+            test_connection.close()
+        except Exception as e:
+            print(f"Cleanup error: {e}")
```

Just let it be raised normally.

---

ok